packages/mantine/src/components/modal/Modal.tsx

```diff
@@ -0,0 +1,61 @@
+import {createStyles, Modal as MantineModal, ModalProps as MantineModalProps} from '@mantine/core';
+import {Children, FunctionComponent, ReactElement, ReactNode} from 'react';
+import {ModalFooter, ModalFooterProps} from './ModalFooter';
+
+const useStyles = createStyles((theme) => {
+    const white = '#fff';
+    return {
+        body: {
+            padding: 0,
+        },
+        modalType: {overflow: 'hidden', width: 550},
+        innerBody: {
+            padding: `${theme.spacing.md}px ${theme.spacing.xl}px ${theme.spacing.lg}px`,
+        },
+        header: {
+            padding: `${theme.spacing.md}px ${theme.spacing.xl}px`,
+            width: '100%',
+            borderBottom: `1px solid ${theme.colors.gray[3]}`,
+            fontSize: theme.headings.sizes.h4.fontSize,
+            lineHeight: theme.headings.sizes.h4.fontSize,
+        },
+        success: {backgroundColor: theme.colors.lime[6], color: white},
+        warning: {backgroundColor: theme.colors.yellow[5], color: white},
+        critical: {
+            backgroundColor: theme.colors.red[6],
+            color: white,
+        },
+        info: {backgroundColor: theme.colors.navy[5], color: white},
+        whiteClose: {color: white, '&:hover': {backgroundColor: 'transparent'}},
+    };
+});
+
+export interface ModalType extends MantineModalProps {
+    type?: 'success' | 'warning' | 'critical' | 'info';
+    Footer?: FunctionComponent<ModalFooterProps>;
+    children: ReactNode;
+}
```

Comment:
The component type and props should be split into 2 different interfaces. Also, lets rename the prop `type` for `variant` since it seems to be more common across Mantine components

```suggestion
export interface ModalProps extends MantineModalProps {
    variant?: 'success' | 'warning' | 'critical' | 'info';
    children: ReactNode;
}

interface ModalType {
    (props: ModalProps): ReactElement;
    Footer: typeof ModalFooter;
}

```